### PR TITLE
Feature/3633 export site api

### LINKED ISSUE #3633: Delete Site - export site
You have the option to export a site and when you click the "export site" button, it will send you to a webview where you can export your site.

### PR DESCRIPTION
**Pull request by kwonye**
Fixes #3633 

screenshot_20160309-141024
screenshot_20160309-141028

To test:
1. Go to Site Settings and click "export content"
2. Click "EXPORT CONTENT"
3. An email will show up with your export zip

Needs review: @oguzkocer @nbradbury

### CODE REVIEW

File: WordPress/build.gradle [hunk 1]
```diff
@@ -35,8 +35,8 @@ android {
         minSdkVersion 14
         targetSdkVersion 23
 
+        vectorDrawables.useSupportLibrary = true
```
Comment (nbradbury) [inline WordPress/build.gradle]: Was this change intentional? We haven't updated to [v23.2](http://android-developers.blogspot.com/2016/02/android-support-library-232.html) of the support library yet, so we're not ready to make this change.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java [hunk 2]
```diff
@@ -702,6 +704,27 @@ public String format(int value) {
...
+        builder.setTitle(R.string.export_your_content);
+        String email = AccountHelper.getDefaultAccount().getEmail();
+        builder.setMessage(getString(R.string.export_your_content_message, email));
+        builder.setPositiveButton(R.string.site_settings_export_content_title, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                exportSite();
+            }
+        });
+        builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java]: Note that using `builder.setNegativeButton(R.string.cancel, null);` here will accomplish the same thing.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java [hunk 3]
```diff
@@ -1187,6 +1210,29 @@ public void onClick(DialogInterface dialog, int which) {
         builder.show();
     }
 
+    private void exportSite() {
+        final Blog currentBlog = WordPress.getCurrentBlog();
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java]: I recommend adding `if (!NetworkUtils.checkConnection(getActivity())) return;` before making the network call. Even though site settings won't appear if there's no connection, there's no guarantee that won't change, and it's possible the connection could exist when settings appears but is lost by the time the user chooses to export.

**Comment by nbradbury** [comment]
With the exception of the `build.gradle` change (which I think is why travis is failing), this looks good code-wise. But the text on the "Export your content" dialog could use some work.

* It's too wordy. I recommend checking out the [writing style page](https://www.google.com/design/spec/style/writing.html) on Google Design.
* "Keep in mind that this content can not be recovered in the future" makes sense on the web because export is part of the delete site process, but here it's separate - so that warning makes it sound like exporting might delete your content.
* Similarly, "...that you can use in the future to resume where you left off" sort of makes sense on the web, but here it's just confusing.

File: WordPress/build.gradle (hunk 1 above)
Comment (maxme) [inline WordPress/build.gradle]: Yep revert this please - ref https://github.com/wordpress-mobile/WordPress-Android/issues/3791

**Comment by oguzkocer** [comment]
I have reviewed the code as well, but nothing to add to Nick's comments. Looks good to me.

File: WordPress/build.gradle (hunk 1 above)
Comment (kwonye) [inline WordPress/build.gradle]: Reverting and moving addition of this line to #3856

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java (hunk 2 above)
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java]: :bow: thanks so much. You have no idea how irked I was I had to do all this just to close the dialog.

**Comment by kwonye** [comment]
> I recommend something much simpler, like just "Your posts, pages, and settings will be emailed to you at %s."

Addressed in df0f442

**Comment by nbradbury** [comment]
Looks good! :shipit:
